cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,270 @@
+package hive
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hiveinternal/v1alpha1"
+	mockk8s "github.com/openshift/osdctl/cmd/hive/clusterdeployment/mock/k8s"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Mock data for ClusterDeployment and ClusterSync that are used across all test cases
+var (
+	cdList = hivev1.ClusterDeploymentList{
+		Items: []hivev1.ClusterDeployment{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example-cluster",
+					Namespace: "uhc-production-1234",
+				},
+				Status: hivev1.ClusterDeploymentStatus{
+					Conditions: []hivev1.ClusterDeploymentCondition{
+						{
+							Type:   "Hibernating",
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+	}
+	csList = v1alpha1.ClusterSyncList{
+		Items: []v1alpha1.ClusterSync{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example-clustersync",
+					Namespace: "uhc-production-1234",
+				},
+				Status: v1alpha1.ClusterSyncStatus{
+					Conditions: []v1alpha1.ClusterSyncCondition{
+						{
+							Type:               "Ready",
+							Status:             corev1.ConditionFalse,
+							Reason:             "Failure",
+							LastTransitionTime: metav1.Time{Time: time.Now()},
+						},
+					},
+					SyncSets: []v1alpha1.SyncStatus{
+						{
+							Name:           "syncset1",
+							Result:         "Failure",
+							FailureMessage: "Failed to sync syncset1",
+						},
+					},
+					SelectorSyncSets: []v1alpha1.SyncStatus{
+						{
+							Name:           "selectorsyncset1",
+							Result:         "Failure",
+							FailureMessage: "Failed to sync selectorsyncset1",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+// Central function to mock the List method and set expectations
+func setupMockClient(mockClient *mockk8s.MockClient, returnErr error) {
+	mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+			switch v := list.(type) {
+			case *hivev1.ClusterDeploymentList:
+				*v = cdList
+			case *v1alpha1.ClusterSyncList:
+				*v = csList
+			}
+			return returnErr
+		}).Times(2) // Expect List to be called twice for ClusterDeployment and ClusterSync
+}
+
+func TestNewCmdClusterSyncFailures(t *testing.T) {
```

Comment:
We don't need to be testing the `cobra` API.

Reply:
I understand your point, and I agree testing the Cobra API might seem redundant. However, since it uses several internal functions, I wrote the test cases for the API itself to avoid writing separate tests for each internal function. This keeps things simpler and more manageable while still covering all the necessary functionality.

Reply:
I still think this should be removed.  If there are other functions that particular to the package, can they be tested with testing the cobra API.